### ISSUE TITLE
Cps1 - cps-b-b15 version

### ISSUE THREAD

**Issue by ragefan** [jwestfall69/mad#3]
Hi,

I compiled the cps1 code and I got a binary for chip 30 and 37. I burnt them on a 27c301 as the original chips and replaced them in the b-board. Nothing happens when starting the board. I may have missed extra steps ?

Edit : just realized my board was a cps-b-15 version...

Thanks

**Comment by jwestfall69** [comment]
Yeah the cps1 stuff is specific to the romset + cpsb version combination.  Thats why there are numerous Makefile files.

what romset are you trying to test?

**Comment by ragefan** [comment]
That's the japanese one with b-15 which I think is :
Street Fighter II: The World Warrior (Japan 910411)                90629B-3   STF29            IOB1  90632C-1     CPS-B-15

I suspect an issue with my b-board and with a ram on the a board. I guess your diagrom would help ☺️

**Comment by jwestfall69** [comment]
I committed an update that adds sf2jf and cps b-15 support.  I don't have that game but it works in mame.  mad diag for cps1 is still pretty basic.  You might be better off using one of the ones here for now https://jammarcade.net/diagnostic-roms/

**Comment by ragefan** [comment]
Thanks a lot for your help ! I really appreciate.
I tested the roms and they are ok. Thanks to them,
I've found I have a DAC issue on the A Board.
All other tests are ok so I validate your roms are working on real hardware ;)

I tried memory viewer but failed to understand what is normal or not. Some memories have ff00 everywhere, some have other things or moving bits, some have 000

![PXL_20240831_075454193](https://github.com/user-attachments/assets/5c1731f7-41e5-4c62-a924-15ab46489ccc)

**Comment by jwestfall69** [comment]
Those lines going through those 2 sets of tiles to me points to it loading the wrong tiles.  So its likely not an issue with the video dac.

For the memory you could compare to mame and see what different.  You might also compare the tile viewer output under the debug menu with what is in mame.

**Comment by ragefan** [comment]
Thanks, I am trying it but not sure how to do it.
The file name of the build is different from the mame dump. Changing the file name to overwrite existing mame files is not working on my side. I shall do it wrong ?

**Comment by ragefan** [comment]
Tested !
There are lot of difference and moving bits in my cpsA reg and cpsB reg. What does that mean ?

**Comment by jwestfall69** [comment]
cpsA and cpsB reg aren't something that will compare well with mame.  Those are the registers of the cpsA and cpsB custom chips.

**Comment by ragefan** [comment]
I discovered I have a issue with the A and the B board by testing with the cps1 diag you mentioned. I tried with the b17 version and a known working B board. The vertical lines we see on the screen before are due to be b-board. I will test all the roms.

The A board still have an issue (look at the sf2 logo on the screen below). (Only part of it is displayed) Unfortunately, nothing that the cps1 diag can detect.

![PXL_20240901_173353062](https://github.com/user-attachments/assets/3af8df60-f594-4b46-b726-4908e38cbcf6)

**Comment by ragefan** [comment]
![Uploading PXL_20240901_173353062.jpg…]()

**Comment by jwestfall69** [comment]
Been a number of month, so going to close this.  Assuming you figured out your issue.
